sherpa/models/basic.py

```diff
@@ -1786,6 +1786,10 @@ class TableModel(ArithmeticModel):
     model can be used (see examples below); interpolation is used if
     the grids of the data and the table do not match.
 
+    When used with an integrated data set (for example,
```

Comment:
May be for a follow-up doc update, but generally with "integrated" or "gridded" or "whatever the term we want to use" the model gets sent both `xlo` and `xhi` - so something like one of

    def evaluator(pars, x, xhi=None):
    def evaluator(pars, x, *args):

[the form depends where in the evaluation chain this happens, I've arbitrarily picked a low-level routine I'd expect to be calling inside the model class]. Does the tablemodel support this or does it only get sent a single array? Or have you not looked into it (the trouble with working on a bit of Sherpa is you then get asked all the questions about it from now on).

For the basic models I tried to add a `Notes` section to explain the difference for point versus grid - e.g.

- https://sherpa.readthedocs.io/en/latest/model_classes/api/sherpa.models.basic.NormGauss1D.html
- https://sherpa.readthedocs.io/en/latest/model_classes/api/sherpa.models.basic.Poisson.html
- https://sherpa.readthedocs.io/en/latest/model_classes/api/sherpa.models.basic.Gauss2D.html

I don't know if we want to formalize this documentation practice (and improve it). I do know there are questions about labelling (eg #1042) which it would be nice to come to some decision on.

Reply:
The table model does not use the `x1` array even if it's given one, hence the "lower edge" description in the docstring. Personally, I could see a new class "IntregratedTableModel" (because of backwards compatibility, I would not want to change this one) that does better, but this PR is about documenting existing features not writing new ones.

Reply:
Hmm. For better or worse we normally have a model support both point (x) and grid (xlo, xhi) options, but that;s for historical raisins. Something to throw on our to-do pile...

Reply:
[going off into the bush here]

I see the evaluation function does accept a `xhi` (aka `x1`) argument

https://github.com/sherpa/sherpa/blob/2e8de205fa88323f6708eca6aecceae76c5953f8/sherpa/models/basic.py#L1827

but it ignores it, and more-importantly there's no way to set such an argument with the loaded data

https://github.com/sherpa/sherpa/blob/2e8de205fa88323f6708eca6aecceae76c5953f8/sherpa/models/basic.py#L1801

Reply:
I think it needs to accept the `xhi` to work with integrated datasets, but then it does not use it - but at least it won't fail and if the bins are small enough and the table model is smooth enough it might just be OK -- or at least that's what I imaging the original intent to be, but that was before my time. Either way it's not easy to be better and at the same time be flux conserving (it currently almost certainly is not). I'll open a new issue about this so we don't forget, but it's certainly not a high-priority item.

Reply:
Ta